src/cosmic_ray/operators/util.py

```diff
@@ -19,3 +19,134 @@ def dec(cls):
         return cls
 
     return dec
+
+
+def dump_node(node, stdout=None):
+    import sys
+    stdout = stdout or sys.stdout
+    write = stdout.write
+
+    def do_dump(node, indent=''):
+        write(f"{indent}{type(node).__name__}({node.type}")
+        value = getattr(node, 'value', None)
+        if value:
+            value = value.replace('\n', '\\n')
+            write(f", '{value}'")
+        children = getattr(node, 'children', None)
+        if children:
+            write(', [\n')
+            for child in children:
+                do_dump(child, indent+' '*4)
+                write(',\n')
+            write(f"{indent}]")
+        write(')')
+        if not indent:
+            write('\n')
+
+    do_dump(node)
+
+
+class ObjTest:
+    """
+    Allowing to navigate into any object and test attribute of any object:
+
+    Examples:
+    >>> ObjTest(node).parent.match(Node, type='node').ok
+
+    Test if node.parent isinstance of Node and node.parent.type == 'node'
+    At each step (each '.' (dot)) you receive an ObjTest object, then
+
+    Navigation:
+    You can call any properties or functions of the base object
+    >>> ObjTest(node).parent.children[2].get_next_sibling()
+
+    Test:
+    >>> ObjTest(node).match(attr='value').match(Class)
+    All in once:
+    >>> ObjTest(node).match(Class, attr='value')
+
+    Conditional navigation:
+    >>> ObjTest(node).IF.match(attr='intermediate').parent.FI
+
+    Final result:
+    >>> ObjTest(node).ok
+    >>> bool(ObjTest(node))
+
+    """
+    def __init__(self, obj):
+        self.obj = obj
+
+    def _clone(self, obj) -> 'ObjTest':
+        return type(self)(obj)
+
+    def match(self, clazz=None, **kwargs) -> 'ObjTest':
```

Comment:
Use `cls` instead of `clazz`. This what we use in almost all of CR.

Reply:
cls is generally classmethod 1st argument, and is the current class. Here it can be any class.

Reply:
I use `cls` in any context where I want to use `class`. If I can't use `cls`, my alternative is `klass` (which I think we use in one place in the CR). For the sake of consistency within this codebase, I'd really prefer to use `cls` here.